### ISSUE TITLE
Build problems with ASYNC_MYSQL disabled

### ISSUE THREAD

**Issue by kristapsk** [facebook/hhvm#7491]
HHVM 3.15 and master is not building with `cmake . -DENABLE_ASYNC_MYSQL=OFF -DENABLE_EXTENSION_ASYNC_MYSQL=OFF`. We want to build it that way because our internal extensions are linked against MySQL++, which is linked against standard libmysql. There are some differences in internal structs between WebScaleSQL and standard libmysql and MySQL++ does not like that (don't remember exact details, colleague was working on this).

Anyway, if such build configuration options are present, they should be working.

Used build command:

```
$ cmake . -DDISABLE_HARDWARE_COUNTERS=ON -DENABLE_ZEND_COMPAT=OFF -DENABLE_EXTENSION_IMAGICK=OFF -DENABLE_ASYNC_MYSQL=OFF -DENABLE_EXTENSION_ASYNC_MYSQL=OFF
```

Steps done to get it to build:
```
$ cd third-party/squangle
$ ln -s src/squangle/mysql_client/ mysql_client
$ ln -s src/squangle/logger/ logger
$ ln -s src/squangle/base/ base
$ cd ../..
$ ln -s third-party/re2/src/re2 re2
```
Seems to be related to https://github.com/hhvm/hhvm-third-party/pull/62.

Also, for the master version, additional changes were needed - https://gist.github.com/kristapsk/1936b187509e9db72f732ca179bfab6b,

**Comment by mofarrell** [comment]
Duplicate of https://github.com/facebook/hhvm/issues/7469

**Comment by kristapsk** [comment]
Fixed with #7925 (not yet merged).
